pytorch_lightning/plugins/training_type/deepspeed.py

```diff
@@ -634,9 +627,17 @@ def _auto_select_batch_size(self):
         batch_size = 1
         train_dl_source = self.lightning_module.trainer._data_connector._train_dataloader_source
         if train_dl_source.is_defined():
-            train_dataloader = train_dl_source.dataloader()
-            if hasattr(train_dataloader, "batch_sampler"):
-                batch_size = train_dataloader.batch_sampler.batch_size
+            try:
+                train_dataloader = train_dl_source.dataloader()
+                if hasattr(train_dataloader, "batch_sampler"):
+                    batch_size = train_dataloader.batch_sampler.batch_size
+            except Exception:
```

Comment:
can't it be entirely resolved with two chained hasattr checks?

Reply:
@carmocca any suggestions on what to narrow it to? I'm uncertain what types it can be.

@awaelchli I'm not sure, `train_dl_source.dataloader()` from my understanding just calls the reference datamodule or hook in the lightning object. This can fail in unexpected ways if `setup` has not been called (Which is not a hard requirement, but is for certain DataModules).

Reply:
You should be able to tell from the stacktrace of the error

Reply:
Sure I could wrap this in `AttributeError` as in this case I try to access an Attribute that doesn't exist, however I think its hard to know what could cause the failure (could be a result of an IO error, Value error etc).

I'd like to be able to fix the case but unfortunately I don't think I can unless someone can guide me here